src/backend/recorders/baseRecorder.ts

```diff
@@ -141,7 +160,19 @@ export abstract class BaseRecorder<T extends WebGLObject> implements IRecorder {
     protected abstract delete(instance: T): number;
 
     protected create(functionInformation: IFunctionInformation): void {
-        // Nothing tracked currently on create.
+        // Removes the spector internal calls to leave only the relevant part.
+        const stackTrace = StackTrace.getStackTrace(5, 0);
+
+        const createdWebGlObject = functionInformation.result as T;
+
+        // Add information where this object was created
+        WebGlObjects.attachWebGlObjectSource(createdWebGlObject, stackTrace);
+
+        // Add our object to the list of tracked objects (object should only exist once).
+        const tag = WebGlObjects.getWebGlObjectTag(createdWebGlObject);
+        this.objects = this.objects || [];
+        if (this.objects[tag.id]) { debugger; }
```

Comment:
This should never happen, so I'll probably remove this line or early out, just to shut up the compiler.
Ideally I'd like to have some sort of assertion or sanity check here.

(A similar case already existed in `updateWithoutSideEffects`, which I consider problematic, too)